### PR TITLE
Add new `Style/FileDirnameLevel` cop

### PR DESCRIPTION
**Pull request by koic**
## Summary

This cop checks for nested `File.dirname`. It replaces repeats with the level argument introduced in Ruby 3.1.

```ruby
# bad
File.dirname(File.dirname(path))

# good
File.dirname(path, 2)
```

See: https://bugs.ruby-lang.org/issues/12194

## Other Information

For `File.expand_path('../..', path)`, there is a partial conflict with `Style/ExpandPathArguments` cop. So this PR does not support it.
https://docs.rubocop.org/rubocop/1.25/cops_style.html#styleexpandpatharguments

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/style/file_dirname_level.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for repeated `File.dirname`.
```
Comment (bbatsov) [inline lib/rubocop/cop/style/file_dirname_level.rb:6]: repeated -> nested

File: lib/rubocop/cop/style/file_dirname_level.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      #
+      # @example
+      #
+      #   # bad
+      #   File.dirname(File.dirname(path))
+      #
+      #   # good
+      #   File.dirname(path, 2)
+      #
+      class FileDirnameLevel < Base
```
Comment (bbatsov) [inline lib/rubocop/cop/style/file_dirname_level.rb:17]: I wonder if `NestedFileDirname` is not a better name, but I'm fine with the proposed name.

File: spec/rubocop/cop/style/file_dirname_level_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Style::FileDirnameLevel, :config do
+  context 'Ruby >= 3.1', :ruby31 do
+    it 'registers and corrects an offense when using `File.dirname(path)` repeated two times' do
```
Comment (bbatsov) [inline spec/rubocop/cop/style/file_dirname_level_spec.rb:5]: nested two times

And so on for the other example descriptions.

File: lib/rubocop/cop/style/file_dirname_level.rb (hunk 2 above)
Comment (koic) [reply lib/rubocop/cop/style/file_dirname_level.rb]: Yeah, I was wondering whether to use that name. So, I've updated to `NestedFileDirname` to be consistent with the doc.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**